### PR TITLE
Move Rust backend to rspc

### PR DESCRIPTION
**Pull request by oscartbeaumont**
Note: This PR requires Rust v1.62

Rust:
 - [x] Using tuple for operation keys
 - [x] Specta generics + `LibraryArgs` abstraction
 - [x] Unit test for exporting bindings cause we can't use the autoexport feature with Tauri
 - [x] Hook up tracing subscriber
 - [x] `CoreEvent`'s working with `NewThumbnail` event
 - [x] Invalidate Query backend
 - [x] Debounce subscriptions
 - [x] Unsubscribe from subscription backend
 - [x] Axum + Tauri error handling + hook the error handling up with `react-query`

Typescript:
 - [x] `useMutation` shouldn't take args at hook creation like it does now.
 - [x] `useLibraryQuery` and `useBridgeQuery`, etc hooked back up with new `LibraryArgs` abstraction
 - [x] Websocket reconnect and exponential backoff in rspc
 - [x] Client based data transformation function on the client
 - [x] Invalidate Query working
 - [x] Unsubscribe from subscription frontend
 - [x] Make subscriptions not reconnection on every render

Final:
 - [x] Fix all Spacedrive Typescript errors
 - [x] Merge conflicts + fix all clippy warnings introduced by PR
 - [x] Upgrade to Prisma Client Rust `0.6.0`
 - [x] Fix Rust warnings and do general code cleanup
 - [x] Error handling -> Remove all `unwrap` and `expect`
 - [x] RSPC disable eslint & prettier
 - [x] RSPC tracing integration -> For errors
 - [x] Spacedrive fix library note updating
 - [x] Generic types working in Specta
 - [x] Specta support for serde flatten
 - [x] Spacedrive `invalidate-query!` should check Specta types are valid
 - [x] Release Specta + RSPC and update imports in Spacedrive

### CODE REVIEW

**Comment by Brendonovich** [comment]
so you're telling me we can have nice things if we build them in-house? crazy

File: core/src/api/invalidate.rs [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+						req.macro_src, req.key
+					);
+				}
+			}
+		}
+	}
+}
+
+#[macro_export]
+macro_rules! invalidate_query {
```
Comment (Brendonovich) [inline core/src/api/invalidate.rs:88]: needs more documentation. the sheer awesomeness of this macro cannot be understood by merely reading its code.

**Review by Brendonovich — DISMISSED** [review]

**Comment by oscartbeaumont** [comment]
@jamiepine This is ready for review and to be merged if you are happy.

**Comment by Brendonovich** [comment]
It should be noted that this PR requires Rust v1.62

**Pull request merged**